restlicsharpclient/restliclient/request/builder/SingleEntityRequestBuilder.cs

```diff
@@ -46,7 +46,7 @@ public SingleEntityRequestBuilder(string baseUrlTemplate) : base(baseUrlTemplate
             input = null;
         }
 
-        public void SetID(TKey id)
+        public virtual void SetID(TKey id)
```

Comment:
If you don't inherit from SingleEntityRequestBuilder for CreateRequestBuilder, this change is not needed.